src/features/defi/providers/cosmos/components/CosmosManager/Claim/ClaimConfirm.tsx

```diff
@@ -0,0 +1,211 @@
+  assetId: AssetId
+  amount?: string
+  contractAddress: string
+  chainId: ChainId
+  onBack: () => void
+}
+
+export const ClaimConfirm = ({
+  assetId,
+  amount,
+  contractAddress,
+  chainId,
+  onBack,
+}: ClaimConfirmProps) => {
+  const [userAddress, setUserAddress] = useState<string>('')
+  const [estimatedGas, setEstimatedGas] = useState<string>('0')
+  const [loading, setLoading] = useState<boolean>(false)
+  const chainAdapterManager = getChainAdapterManager()
+  const { state: walletState } = useWallet()
+  const translate = useTranslate()
+  const claimAmount = bnOrZero(amount).toString()
+  const history = useHistory()
+
+  // Asset Info
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace: 'slip44',
+    assetReference: ASSET_REFERENCE.Cosmos, // TODO: Programmatic
+  })
+  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId))
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  const toast = useToast()
+
+  const { handleStakingAction } = useStakingAction()
+
+  const handleConfirm = async () => {
+    if (!walletState.wallet || !contractAddress || !userAddress) return
+    setLoading(true)
+
+    const { gasLimit, gasPrice } = await getStakingFees(asset, feeMarketData.price)
+
+    try {
+      const broadcastTxId = await handleStakingAction({
+        asset,
+        validator: contractAddress,
+        chainSpecific: {
+          gas: gasLimit,
+          fee: bnOrZero(gasPrice).times(`1e+${asset?.precision}`).toString(),
+        },
+        value: bnOrZero(claimAmount).times(`1e+${asset.precision}`).toString(),
+        action: StakingAction.Claim,
+      })
+      history.push('/status', {
+        txid: broadcastTxId,
+        assetId,
+        amount,
+        userAddress,
+        estimatedGas,
+        chainId,
+      })
+    } catch (error) {
+      console.error('ClaimWithdraw:handleConfirm error', error)
+      toast({
+        position: 'top-right',
+        description: translate('common.transactionFailedBody'),
+        title: translate('common.transactionFailed'),
+        status: 'error',
+      })
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!walletState.wallet) return
+
+        const { gasLimit, gasPrice } = await getStakingFees(asset, feeMarketData.price)
+
+        const chainAdapter = chainAdapterManager.get(chainId) as unknown as
+          | cosmossdk.cosmos.ChainAdapter
+          | cosmossdk.osmosis.ChainAdapter
+        const userAddress = await chainAdapter.getAddress({ wallet: walletState.wallet })
+        setUserAddress(userAddress)
+        const gasEstimate = bnOrZero(gasPrice).times(gasLimit).toFixed(0)
+        setEstimatedGas(gasEstimate)
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('CosmosClaim error:', error)
```

Comment:
moduleLogger

Reply:
Also from the Foxy implementation, will add it in the same stacked PR

Reply:
https://github.com/shapeshift/web/pull/2323